src/DeductionManager.cpp

```
@@ -116,6 +116,38 @@ static float applyOpenTag3D(const char* uid, float pending) {
+    const char* skipReason = NULL;
+    if (!opentag3d_can_encode(ot3d.tag_version)) {
+        skipReason = "newer minor format";
+    } else if (opentag3d_major(ot3d.tag_version) >= 2 && ot3d.measured_filament_weight_g == 0) {
+        skipReason = "v2 tag has no measured weight (target weight is nominal)";
+    }
+    if (skipReason) {
+        Serial.printf("DeductionManager: OpenTag3D %s (v%u) — %s; tag left untouched\n",
+                      uid, ot3d.tag_version, skipReason);
+        if (SpoolmanManager::getInstance().isConfigured()) {
+            float spDeducted = SpoolmanManager::getInstance().deductFromSpoolman(uid, pending);
+            if (spDeducted > 0.0f) {
```

Clang (14.0.6)

[warning] floating point literal has suffix 'f', which is not uppercase

(readability-uppercase-literal-suffix)